enhancements/network/cloud-egress-ip.md

```diff
@@ -0,0 +1,468 @@
+node objects according to a defined convention which the network plugins can
+read. That information would then be taken into account during the plugin's
+assignment procedure.
+
+Vice versa, the network plugin could issue a "cloud egress IP assignment
+request" by means of annotating the node object. The cloud component could use
+this annotation as input to start executing an IP assignment in the cloud.
+
+Given the minimal information needed to be conveyed to the network plugin from
+the cloud component, and the dependencies that a CRD would introduce: it is
+estimated that the annotated model might be preferred.
+
+##### Cloud credentials operator - CCO
+
+The cloud credentials operator (CCO) [10] is an OpenShift component that manages
+the `CredentialRequest` CR. This resource allows other components access to the
+cloud API by creating a secret in the `kube-system` namespace which the resource
+issuing the `CredentialRequest` can fetch, containing the permissions required
+to communicate with the cloud API by the requesting component. This cloud
+component would thus need to mount a secret from the location the CCO generates
+its output. If a upstream model is chosen there would be an added requirement on
+the cluster admin to generate such a secret manually with the proper
+credentials.
+
+[10]: https://github.com/openshift/cloud-credential-operator
+
+##### RBAC
+
+Given the approach chosen above, a separate set of RBAC permissions will be
+required for the cloud component / network plugins. These are listed below.
+
+Independently of the model chosen, the cloud component will require:
+
+WATCH/LIST/GET:
+
+- `secrets` - to watch for the creation of cloud credentials by the CCO
+- `node` - presumably to be able to verify node names / other info when
+  communicating with the cloud API
+
+**CRD**
+
+Using a CRD based approach we would need the following RBAC permissions:
+
+Network plugin - WATCH/LIST/GET
+
+- Cloud component CRD - as to retrieve the node subnet information from the
+  cloud API / issue cloud egress IP assignment, using the CRD.
+
+Cloud component - UPDATE/PATCH/CREATE
+
+- Cloud component CRD - as to update its CR with the status of the cloud
+  assignment
+
+**annotated approach**
+
+Network plugin - UPDATE/PATCH/CREATE
+
+- `node` - as to annotate nodes with the cloud egress IP assignment to be read
+  by the cloud component
+
+Cloud component - UPDATE/PATCH/CREATE
+
+- `node` - as to annotate nodes with the subnet information read by the network
+  plugin / notify the user of a successful/unsuccessful cloud assignment
+- `events` - possibly as to notify the user of successful/unsuccessful cloud
+  assignment
+
+##### Deployment model
+
+The following section describes a deployment model assuming the simplest
+approach possible: keeping the cloud component as a dedicated OpenShift
+component and using a annotated approach.
+
+###### OVN-Kubernetes
+
+**Day 0**
```

Comment:
Steps 1, 3, and 4 are things that happen whenever a new node is added to the cluster, not just at install time. I don't think the "day 0" / "day 1/2" split is the right way to describe this

Reply:
I will re-formulate